core/trino-main/src/test/java/io/trino/util/TestMetadataQueriesQueryType.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.util;
+
+import io.trino.Session;
+import io.trino.SessionTestUtils;
+import io.trino.execution.QueryPreparer;
+import io.trino.execution.QueryPreparer.PreparedQuery;
+import io.trino.sql.parser.SqlParser;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.Optional;
+
+import static io.trino.spi.resourcegroups.QueryType.DESCRIBE;
+import static io.trino.spi.resourcegroups.QueryType.SELECT;
+import static io.trino.util.StatementUtils.getQueryType;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.testng.Assert.assertEquals;
+
+@TestInstance(PER_CLASS)
+public class TestMetadataQueriesQueryType
+{
+    private Session session;
+    private QueryPreparer queryPreparer;
+
+    @BeforeAll
+    public void init()
+    {
+        session = SessionTestUtils.TEST_SESSION;
```

Comment:
Please add tests with default schema and catalog options

Reply:
Done.  I also added a test for a query with a polymorphic table function.